src/tools/ceph_objectstore_tool.cc

```diff
@@ -646,23 +646,23 @@ int do_trim_pg_log(ObjectStore *store, const coll_t &coll,
 
   cerr << "Log bounds are: " << "(" << info.log_tail << ","
        << info.last_update << "]" << std::endl;
-
   uint64_t max_entries = g_ceph_context->_conf->osd_max_pg_log_entries;
-  if (info.last_update.version - info.log_tail.version <= max_entries) {
-    cerr << "Log not larger than osd_max_pg_log_entries " << max_entries << std::endl;
-    return 0;
-  }
 
+  PGLog::IndexedLog pg_log;
+  pg_log.tail = info.log_tail;
+  pg_log.head = info.last_update;
+  pg_log.skip_can_rollback_to_to_head();
+  
   ceph_assert(info.last_update.version > max_entries);
```

Comment:
I understand why you removed the check. My point is that after its removal, the `ceph_assert(info.last_update.version > max_entries)` may fail. And we still need the condition `info.last_update.version > max_entries` to be valid (i.e. we can't just remove the assert), because we use `info.last_update.version - max_entries` below. So my idea is that we should set `max_entries` not just to `conf->osd_max_pg_log_entries` but to something like `min(conf->osd_max_pg_log_entries, info.last_update.version - 1)` (I have not checked though if `last_update.version` may be 0 though).

Reply:
@trociny i done it differently, please review